clang/lib/Parse/ParseDecl.cpp

```diff
@@ -7279,6 +7303,10 @@ void Parser::ParseParameterDeclarationClause(
         }
       }
 
+      // If a where clause is declared, attached it to the parameter
+      // declaration.
+      AttachWhereClause(Param);
```

Comment:
The parsing and semantic analysis of bounds declarations on function parameters are done in a deferred manner - see comment on lines 7095 to 7099 and the code on lines 7350 to 7359 - probably because bounds declarations may refer to function parameters that may occur later in the parameter list. This may apply to where clauses also - this needs to be looked into.